### PR TITLE
*: fix two error shade bug.

### PR DESCRIPTION
**Pull request by hicqu**
In some old code, variables could be shaded by others. It's a bad code style, and maybe causes bugs. This PR fix it.

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

**Comment by hicqu** [comment]
The json decode bug is fixed. @jackysp , PTAL, thanks!

File: util/types/json/serdes.go [hunk 1]
```diff
@@ -229,8 +229,10 @@ func encodeJSONString(s string, buffer *bytes.Buffer) {
 }
 
 func decodeJSONString(s *string, data []byte) (err error) {
+	var length uint64
 	var reader = bytes.NewReader(data)
-	if length, err := binary.ReadUvarint(reader); err == nil {
+	length, err = binary.ReadUvarint(reader)
+	if err == nil && length > 0 {
```
Comment (jackysp) [inline util/types/json/serdes.go]: so "" will return length > 0?

Comment (hicqu) [inline util/types/json/serdes.go]: If you parse `"\""\"""` into JSON as `j`, `j.s` will be an empty string with length = 0.

**Pull request merged**